oauth2_http/javatests/com/google/auth/oauth2/MockTokenServerTransport.java

```diff
@@ -206,7 +210,8 @@ public LowLevelHttpResponse execute() throws IOException {
             }
             String foundSecret = query.get("client_secret");
             String expectedSecret = clients.get(foundId);
-            if (foundSecret == null || !foundSecret.equals(expectedSecret)) {
+            if ((foundSecret == null || !foundSecret.equals(expectedSecret))
+                && clients.get("Authorization") == null) {
```

Comment:
Can you pull this condition into a variable? It's getting hard to grok

Reply:
Do it through the way of in-house authorization way.